lib/mayaHydra/mayaPlugin/renderOverride.cpp

```diff
@@ -611,15 +625,34 @@ MStatus MtohRenderOverride::Render(
         }
     }
 
-    const auto      displayStyle = drawContext.getDisplayStyle();
+    const unsigned int currentDisplayStyle = drawContext.getDisplayStyle();
     MayaHydraParams delegateParams = _globals.delegateParams;
-    delegateParams.displaySmoothMeshes = !(displayStyle & MHWRender::MFrameContext::kFlatShaded);
+    delegateParams.displaySmoothMeshes = !(currentDisplayStyle & MHWRender::MFrameContext::kFlatShaded);
 
     if (_mayaHydraSceneIndex) {
         _mayaHydraSceneIndex->SetDefaultLightEnabled(_hasDefaultLighting);
         _mayaHydraSceneIndex->SetDefaultLight(_defaultLight);
         _mayaHydraSceneIndex->SetParams(delegateParams);
         _mayaHydraSceneIndex->PreFrame(drawContext);
+
+        if (NeedToRecreateTheSceneIndicesChain(currentDisplayStyle, _oldDisplayStyle)){
+            //We need to recreate the filtering scene index chain after the merging scene index as there was a change such as in the BBox display style which has been turned on or off.
+            _lastFilteringSceneIndexBeforeCustomFiltering = nullptr;//Release
+            _CreateSceneIndicesChainAfterMergingSceneIndex(drawContext);
+            auto& manager = Fvp::ViewportInformationAndSceneIndicesPerViewportDataManager::Get();
+            manager.RemoveViewportInformation(std::string(panelName.asChar()));
+            //Get information from viewport
+            std::string cameraName;
+            M3dView view;
+            if (M3dView::getM3dViewFromModelPanel(panelName, view)){
+                MDagPath dpath;
+                view.getCamera(dpath);
+                MFnCamera viewCamera(dpath);
+                cameraName = viewCamera.name().asChar();
+            }
+            const Fvp::InformationInterface::ViewportInformation hydraViewportInformation(std::string(panelName.asChar()), cameraName);
+            manager.AddViewportInformation(hydraViewportInformation, _renderIndexProxy, _lastFilteringSceneIndexBeforeCustomFiltering);
+        }
```

Comment:
Why do we need to re-create the scene index chain?  Can't we just dirty the whole scene and redraw with the new style?

Reply:
There is 2 possibilitites here.
1) We always have a BBox scene index with a flag to enable/disable it. 
2) we remove/add the BBox scene index when needed so we recreate the chain. 
In the near future, we'll have some more cases like override the scene with the default material which will likely be another scene index. 
I think it's better to completely remove the useless scene indices (so using solution 2) from above) instead of keeping them in the chain and disabling them as it will take some indirection for nothing.

Reply:
O.K., that's a valid choice.  Can we add this as an architectural-level code comment in the code, and possibly in a Markdown file somewhere, that this is what we expect in terms of architecture?